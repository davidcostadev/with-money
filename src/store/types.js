export const ADD_USER = 'ADD_USER';
export const TO_CURRENT_MONTH = 'TO_CURRENT_MONTH';
export const TO_PREVIUS_MONTH = 'TO_PREVIUS_MONTH';
export const TO_NEXT_MONTH = 'TO_NEXT_MONTH';
