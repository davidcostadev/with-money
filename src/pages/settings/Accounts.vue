<template>
  <div class="full-height">
    <md-toolbar>
      <router-link to="/settings">
        <md-button class="md-icon-button">
          <md-icon>keyboard_backspace</md-icon>
        </md-button>
      </router-link>
      <h3 class="md-title">Contas</h3>
    </md-toolbar>
    <md-content>
      <md-list v-for="account in accounts" :key="account.id">
        <md-list-item :to="`/settings/accounts/edit/${account.id}`">
          <span>{{account.name}}</span>
          <account-icon :value="account.type" />
        </md-list-item>
      </md-list>
    </md-content>

    <md-button
      class="md-fab md-primary menu-speed"
      to="/settings/accounts/new"
    >
      <md-icon>note_add</md-icon>
    </md-button>
  </div>
</template>

<script>
import Accounts from '../../services/accounts';
import AccountIcon from '../../components/AccountIcon';

export default {
  data() {
    return {
      accounts: [],
    };
  },
  components: { AccountIcon },
  methods: {
    async getAccounts() {
      const { data } = await Accounts.list();

      this.accounts = data;
    },
  },
  created() {
    this.getAccounts();
  },
};
</script>
