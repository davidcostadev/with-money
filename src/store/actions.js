import * as types from './types';

export default {
  addUser({ commit }, payload) {
    commit(types.ADD_USER, payload);
  },
  toNextMonth({ commit }) {
    commit(types.TO_NEXT_MONTH);
  },
  toCurrentMonth({ commit }) {
    commit(types.TO_CURRENT_MONTH);
  },
  toPreviousMonth({ commit }) {
    commit(types.TO_PREVIUS_MONTH);
  },
};
