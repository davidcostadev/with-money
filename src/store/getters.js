export default {
  user(state) {
    return state.user;
  },
  current_month(state) {
    return state.current_month;
  },
  state_month(state) {
    return state.state_month;
  },
  state_month_str(state) {
    return state.state_month_str;
  },
  next_month_str(state) {
    return state.next_month_str;
  },
  previous_month_str(state) {
    return state.previous_month_str;
  },
};
