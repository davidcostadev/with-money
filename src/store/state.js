import Moment from 'moment';

export default {
  user: null,
  current_month: Moment(),
  state_month: Moment(),
  state_month_str: Moment().format('MMM YY'),
  next_month_str: Moment().add(1, 'month').format('MMM YY'),
  previous_month_str: Moment().subtract(1, 'month').format('MMM YY'),
  flash_message: '',
};
