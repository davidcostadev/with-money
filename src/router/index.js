import Vue from 'vue';
import Router from 'vue-router';
import Login from '@/pages/Login';
import TransactionEdit from '@/pages/TransactionEdit';
import TransactionList from '@/pages/TransactionList';
import store from '../store';

Vue.use(Router);

const insecure = (to, from, next) => {
  const token = window.localStorage.getItem('token');

  if (!store.getters.user && !token) {
    next();
    return;
  }

  next('/');
};

const secure = (to, from, next) => {
  const token = window.localStorage.getItem('token');

  if (store.getters.user || token) {
    next();
    return;
  }

  next('/login');
};

export default new Router({
  routes: [
    {
      path: '/login',
      name: 'login',
      component: Login,
      beforeEnter: insecure,
    },
    {
      path: '/',
      name: 'TransactionList',
      component: TransactionList,
      beforeEnter: secure,
    },
    {
      path: '/transaction/:id',
      name: 'TransactionEdit',
      component: TransactionEdit,
      beforeEnter: secure,
    },
  ],
});
