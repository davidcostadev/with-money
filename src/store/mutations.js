import Moment from 'moment';
import * as types from './types';
import { updateStatesMont } from '../utils/state';

export default {
  [types.ADD_USER](state, payload) {
    state.user = payload;
  },
  [types.TO_CURRENT_MONTH](state) {
    state.state_month = state.current_month;
    updateStatesMont(state);
  },
  [types.TO_PREVIUS_MONTH](state) {
    state.state_month = Moment(state.state_month).subtract(1, 'month');
    updateStatesMont(state);
  },
  [types.TO_NEXT_MONTH](state) {
    state.state_month = Moment(state.state_month).add(1, 'month');
    updateStatesMont(state);
  },
};
