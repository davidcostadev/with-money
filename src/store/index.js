import Vue from 'vue';
import Vuex from 'vuex';
import Moment from 'moment';
import mutations from './mutations';
import actions from './actions';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    user: null,
    current_month: Moment(),
    state_month: Moment(),
    state_month_str: Moment().format('MMMM YY'),
    next_month_str: Moment().add(1, 'month').format('MMMM YY'),
    previous_month_str: Moment().subtract(1, 'month').format('MMMM YY'),
  },
  getters: {
    user(state) {
      return state.user;
    },
    current_month(state) {
      return state.current_month;
    },
    state_month(state) {
      return state.state_month;
    },
    state_month_str(state) {
      return state.state_month_str;
    },
    next_month_str(state) {
      return state.next_month_str;
    },
    previous_month_str(state) {
      return state.previous_month_str;
    },
  },
  mutations,
  actions,
});
