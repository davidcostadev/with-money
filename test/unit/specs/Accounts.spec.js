import { createLocalVue, shallowMount, mount } from '@vue/test-utils';
import VueMaterial from 'vue-material';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import Accounts from '../../../src/pages/settings/Accounts';
import store from '../../../src/store';
import router from '../../../src/router';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueMaterial);
localVue.use(VueRouter);

const accounts = [
  {
    id: 2,
    UserId: 1,
    name: 'Carteira',
    type: 'wallet',
    initalValue: '650.00',
    createdAt: '2018-08-04T14:33:21.000Z',
    updatedAt: '2018-08-04T14:33:21.000Z',
  },
  {
    id: 1,
    UserId: 1,
    name: 'Banco Inter',
    type: 'checking_account',
    initalValue: '505.18',
    createdAt: '2018-08-04T14:33:21.000Z',
    updatedAt: '2018-08-04T14:33:21.000Z',
  },
];

const mockAccounts = {
  data: accounts,
  pagination: {
    totalItems: 2,
    currentPage: 1,
    perPage: 100,
    totalPages: 1,
    nextPage: null,
    previousPage: null,
  },
};


jest.mock('../../../src/services/accounts', () => ({
  list: () => Promise.resolve(mockAccounts),
}));

describe('Accounts Component', () => {
  beforeAll(() => {
    global.localStorage = {
      getItem: jest.fn().mockReturnValue(JSON.stringify({
        id: '1',
        name: 'name',
      })),
    };
  });

  describe('Render', () => {
    it('default', () => {
      const wrapper = shallowMount(Accounts, { store, localVue, router });

      expect(wrapper.isVueInstance).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });

    it('with accounts', () => {
      const wrapper = mount(Accounts, { store, localVue, router });
      wrapper.setData({
        accounts,
      });

      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
