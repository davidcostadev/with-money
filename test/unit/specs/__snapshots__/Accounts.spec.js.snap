// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accounts Component Render default 1`] = `
<div
  class="full-height"
>
  <mdtoolbar-stub
    class="toolbar"
  >
    <router-link-stub>
      <mdbutton-stub
        class="md-icon-button"
      >
        <mdicon-stub>
          keyboard_backspace
        </mdicon-stub>
      </mdbutton-stub>
    </router-link-stub>
     
    <h3
      class="md-title"
    >
      Contas
    </h3>
  </mdtoolbar-stub>
   
  <div
    class="content-scroll"
  >
    <mdcontent-stub />
     
    <mdbutton-stub
      class="md-fab md-primary menu-speed"
    >
      <mdicon-stub>
        note_add
      </mdicon-stub>
    </mdbutton-stub>
  </div>
</div>
`;

exports[`Accounts Component Render with accounts 1`] = `
<div
  class="full-height"
>
  <div
    class="md-toolbar toolbar md-theme-default md-elevation-4"
  >
    <a
      class=""
      href="#/settings"
    >
      <button
        class="md-button md-icon-button md-theme-default"
        type="button"
      >
        <div
          class="md-ripple"
        >
          <div
            class="md-button-content"
          >
            <i
              class="md-icon md-icon-font md-theme-default"
            >
              keyboard_backspace
            </i>
          </div>
           
        </div>
      </button>
    </a>
     
    <h3
      class="md-title"
    >
      Contas
    </h3>
  </div>
   
  <div
    class="content-scroll"
  >
    <div
      class="md-content md-theme-default"
    >
      <ul
        class="md-list md-theme-default"
      >
        <li
          class="md-list-item"
          to="/settings/accounts/edit/2"
        >
          <a
            class="md-list-item-router md-list-item-container md-button-clean"
            href="#/settings/accounts/edit/2"
            mdripple="true"
          >
            <div
              class="md-list-item-content md-ripple"
            >
              <span>
                Carteira
              </span>
               
              <i
                class="md-icon md-icon-font md-theme-default"
              >
                account_balance_wallet
              </i>
               
            </div>
          </a>
        </li>
      </ul>
      <ul
        class="md-list md-theme-default"
      >
        <li
          class="md-list-item"
          to="/settings/accounts/edit/1"
        >
          <a
            class="md-list-item-router md-list-item-container md-button-clean"
            href="#/settings/accounts/edit/1"
            mdripple="true"
          >
            <div
              class="md-list-item-content md-ripple"
            >
              <span>
                Banco Inter
              </span>
               
              <i
                class="md-icon md-icon-font md-theme-default"
              >
                account_balance
              </i>
               
            </div>
          </a>
        </li>
      </ul>
    </div>
     
    <a
      class="md-button md-fab md-primary menu-speed md-theme-default"
      href="#/settings/accounts/new"
    >
      <div
        class="md-ripple"
      >
        <div
          class="md-button-content"
        >
          <i
            class="md-icon md-icon-font md-theme-default"
          >
            note_add
          </i>
        </div>
         
      </div>
    </a>
  </div>
</div>
`;
